pages/api/threads/[id].ts

```
@@ -0,0 +1,92 @@
+
+import { NextApiRequest, NextApiResponse } from 'next';
+import nc from 'next-connect';
+import { onError, onNoMatch, requireUser } from 'lib/middleware';
+import { withSessionRoute } from 'lib/session/withSession';
+import { prisma } from 'db';
+import { computeUserPagePermissions } from 'lib/permissions/pages/page-permission-compute';
+
+const handler = nc<NextApiRequest, NextApiResponse>({ onError, onNoMatch });
+
+handler.use(requireUser)
+  .delete(deleteThread)
+  .put(updateThread);
```

add requireKeys(['resolved'], 'query')

---

Good point. Better to return an appropriate error response than the generic 'Something went wrong'